tests/Integration/Momento.Sdk.Tests/TestCacheClient.cs

```diff
@@ -0,0 +1,374 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Momento.Sdk.Auth;
+using Momento.Sdk.Config;
+using Momento.Sdk.Requests;
+
+namespace Momento.Sdk.Tests;
+
+/// <summary>
+/// Client call wrapper to make test expectations easier to write.
+/// </summary>
+public class TestCacheClient : ICacheClient
+{
+    private readonly CacheClient client;
+
+    public TestCacheClient(IConfiguration config, ICredentialProvider authProvider, TimeSpan defaultTtl)
+    {
+        client = new CacheClient(config, authProvider, defaultTtl);
+    }
+
+    /// <summary>
+    /// Wait for the service to settle any slow replication or intermediate response cache.
+    /// Momento doesn't have strict read-after-write consistency guarantees, but a little
+    /// time should settle it in normal situations.
+    /// </summary>
+    /// <returns>task with action's result executed after the quiesce period</returns>
+    private Task<T> QuiesceAndThenRun<T>(Func<Task<T>> action) {
+        return ((Func<Task<T>>)(async () =>
+        {
+            await Task.Delay(TimeSpan.FromMilliseconds(10));
+            return await action();
+        }))();
+    }
+
+    public Task<CreateCacheResponse> CreateCacheAsync(string cacheName)
+    {
+        return ((ICacheClient)client).CreateCacheAsync(cacheName);
+    }
```

Comment:
As mentioned in the pr description, these methods are all generated by visual studio.

Reply:
why is that cast necessary tho

Reply:
> these methods are all generated by visual studio.

probably not necessary, but this is what visual studio will spew out for the next method you tell it to override. So I just figured this would be the lowest friction thing to do - let the code generator generate code.